File: frontend/src/models/recipe/RecipeIngredientModel.ts
import type { NutritionValuesModel } from "./NutritionValuesModel.ts";
import type { Units } from "./Units.ts";

export type RecipeIngredientModel = {
  id: string;
  name: string;
  amount: number;
  unit: Units;
  nutritionValues: NutritionValuesModel;
  allergens: string[] | undefined;
  hint?: string | null;
};

File: backend/src/main/java/org/example/backend/recipe/model/RecipeIngredientModel.java
package org.example.backend.recipe.model;

import com.mongodb.lang.Nullable;
import org.springframework.data.mongodb.core.mapping.Field;

public record RecipeIngredientModel(
        @Field("id") String id,
        String name,
        double amount,
        String unit,
        NutritionValuesModel nutritionValues,
        @Nullable String[] allergens,
        @Nullable String hint
) {
}
